src/goto-instrument/cover_instrument_assume.cpp

```diff
@@ -0,0 +1,52 @@
+/// \file cover_instrument_assume.cpp
+/// Coverage Instrumentation for ASSUME instructions.
+
+#include "cover_instrument.h"
+
+#include "ansi-c/expr2c.h"
+#include "goto-programs/goto_program.h"
+#include "util/std_expr.h"
+#include <util/namespace.h>
+
+/// Instrument program to check coverage of assume statements.
+/// \param goto_program The goto-program (function under instrumentation).
+/// \param i_it The current instruction (instruction under instrumentation).
+/// \param make_assertion The assertion generator function.
+void cover_assume_instrumentert::instrument(
+  const irep_idt &,
+  goto_programt &goto_program,
+  goto_programt::targett &i_it,
+  const cover_blocks_baset &,
+  const assertion_factoryt &make_assertion) const
+{
+  namespacet ns{symbol_tablet()};
+  if(i_it->is_assume())
+  {
+    // Get assume-instruction location.
+    const auto assume_condition = expr2c(i_it->get_condition(), ns);
+    auto location = i_it->source_location;
+
+    // Add appropriate comment to new assert(false) instruction before assume.
+    location.set_comment(
+      "assert(false) before assume(" + assume_condition + ")");
+    const auto assert_before = make_assertion(false_exprt{}, location);
+    goto_program.insert_before(i_it, assert_before);
+
+    // Add appropriate comment to new assert(false) instruction after assume.
+    location.set_comment(
+      "assert(false) after assume(" + assume_condition + ")");
+    const auto assert_after = make_assertion(false_exprt{}, location);
+    goto_program.insert_after(i_it, assert_after);
+  }
+  // Otherwise, skip existing assertions.
+  else if(i_it->is_assert())
+  {
+    const auto location = i_it->source_location;
+    const auto comment = std::string(location.get_comment().c_str());
+    // Filter based on if assertion was added by us as part of instrumentation.
+    if(comment.find("assert(false) ") == std::string::npos)
```

Comment:
🚫 use `initialize_source_locations()` which sets `source_location.set_property_class` to `coverage` and check that.

Reply:
Done 👍🏻